pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -1438,99 +1473,52 @@ class Timedelta(_Timedelta):
     We see that either way we get the same result
     """
 
-    _req_any_kwargs_new = {"weeks", "days", "hours", "minutes", "seconds",
-                           "milliseconds", "microseconds", "nanoseconds"}
+    _allowed_kwargs = (
+        "weeks", "days", "hours", "minutes", "seconds", "milliseconds", "microseconds", "nanoseconds"
+    )
 
     def __new__(cls, object value=_no_input, unit=None, **kwargs):
-        cdef _Timedelta td_base
+        cdef:
+            _Timedelta td_base
+            NPY_DATETIMEUNIT out_reso = NPY_FR_ns
 
+        # process kwargs iff no value passed
         if value is _no_input:
-            if not len(kwargs):
-                raise ValueError("cannot construct a Timedelta without a "
-                                 "value/unit or descriptive keywords "
-                                 "(days,seconds....)")
-
-            kwargs = {key: _to_py_int_float(kwargs[key]) for key in kwargs}
-
-            unsupported_kwargs = set(kwargs)
-            unsupported_kwargs.difference_update(cls._req_any_kwargs_new)
-            if unsupported_kwargs or not cls._req_any_kwargs_new.intersection(kwargs):
+            if not kwargs:
+                raise ValueError(
+                    "cannot construct a Timedelta without a value/unit "
+                    "or descriptive keywords (days,seconds....)"
+                )
+            if not kwargs.keys() <= set(cls._allowed_kwargs):
                 raise ValueError(
                     "cannot construct a Timedelta from the passed arguments, "
-                    "allowed keywords are "
-                    "[weeks, days, hours, minutes, seconds, "
-                    "milliseconds, microseconds, nanoseconds]"
+                    f"allowed keywords are {cls._allowed_kwargs}"
                 )
-
-            # GH43764, convert any input to nanoseconds first and then
-            # create the timestamp. This ensures that any potential
-            # nanosecond contributions from kwargs parsed as floats
-            # are taken into consideration.
-            seconds = int((
+            # GH43764, convert any input to nanoseconds first, to ensure any potential
+            # nanosecond contributions from kwargs parsed as floats are included
+            kwargs = collections.defaultdict(int, {key: _to_py_int_float(val) for key, val in kwargs.items()})
```

Comment:
is there a compelling reason to use defaultdict instead of .get?

Reply:
The motivation was stylistic: repeating `get[component]` is slightly cleaner than `get(component, 0)`. Don't think that counts as a compelling though 😄

The last perf test I ran didn't show any appreciable slowdown for creating a `Timedelta` from components, but I don't feel strongly about this, so happy to keep the old style if desired.

Reply:
id prefer the old style, yes.  my expectation is that cython can optimize the .get to pure-C but not the defaultdict stuff

Reply:
> my expectation is that cython can optimize the .get to pure-C but not the defaultdict stuff

Good call, that seems to be the case:

```
       before           after         ratio
     [7332c1c1]       [f31daa88]
     <td-construction~2>       <td-construction~1>
-      5.93±0.2μs      4.34±0.02μs     0.73  tslibs.timedelta.TimedeltaConstructor.time_from_components
```